File: go-sdk/pkg/models/operations/getabranchpassword.go
package operations

type GetABranchPasswordPathParams struct {
	Branch       string `pathParam:"style=simple,explode=false,name=branch"`
	Database     string `pathParam:"style=simple,explode=false,name=database"`
	ID           string `pathParam:"style=simple,explode=false,name=id"`
	Organization string `pathParam:"style=simple,explode=false,name=organization"`
}

type GetABranchPasswordQueryParams struct {
	ReadOnlyRegionID *string `queryParam:"style=form,explode=true,name=read_only_region_id"`
}

type GetABranchPasswordRequest struct {
	PathParams  GetABranchPasswordPathParams
	QueryParams GetABranchPasswordQueryParams
}

type GetABranchPasswordResponse struct {
	ContentType                                string
	StatusCode                                 int
	GetABranchPassword200ApplicationJSONObject map[string]map[string]interface{}
}

File: ts-sdk/src/sdk/models/operations/getabranchpassword.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import { Type } from "class-transformer";


export class GetABranchPasswordPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=branch" })
  branch: string;

  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=database" })
  database: string;

  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=id" })
  id: string;

  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=organization" })
  organization: string;
}

export class GetABranchPasswordQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=read_only_region_id" })
  readOnlyRegionId?: string;
}

export class GetABranchPasswordRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: GetABranchPasswordPathParams;

  @SpeakeasyMetadata()
  queryParams: GetABranchPasswordQueryParams;
}

export class GetABranchPasswordResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  getABranchPassword200ApplicationJSONObject?: Record<string, Record<string, any>>;
}